net/local/local_recvmsg.c

```diff
@@ -375,30 +511,48 @@ psock_dgram_recvmsg(FAR struct socket *psock, FAR struct msghdr *msg,
 ssize_t local_recvmsg(FAR struct socket *psock, FAR struct msghdr *msg,
                       int flags)
 {
+  ssize_t readlen;
+
   DEBUGASSERT(psock && psock->s_conn && msg);
 
   /* Check for a stream socket */
 
 #ifdef CONFIG_NET_LOCAL_STREAM
   if (psock->s_type == SOCK_STREAM)
     {
-      return psock_stream_recvmsg(psock, msg, flags);
+      readlen = psock_stream_recvmsg(psock, msg, flags);
     }
   else
 #endif
 
 #ifdef CONFIG_NET_LOCAL_DGRAM
   if (psock->s_type == SOCK_DGRAM)
     {
-      return psock_dgram_recvmsg(psock, msg, flags);
+      readlen = psock_dgram_recvmsg(psock, msg, flags);
     }
   else
 #endif
     {
       DEBUGPANIC();
       nerr("ERROR: Unrecognized socket type: %" PRIu8 "\n", psock->s_type);
-      return -EINVAL;
+      readlen = -EINVAL;
     }
+
+#ifdef CONFIG_NET_LOCAL_SCM
+  /* Receive the control message */
+
+  if (readlen >= 0 && msg->msg_control &&
+      msg->msg_controllen > sizeof(struct cmsghdr))
+    {
+      int ret = local_recvctl(psock->s_conn, msg);
+      if (ret < 0)
```

Comment:
since local_recvctl is called inside local_recvmsg, we don't need modify psock_dgram_recvmsg/psock_stream_recvmsg naming and prototype